Source/WebCore/html/shadow/TextControlInnerElements.cpp

```diff
@@ -254,13 +254,13 @@ std::optional<Style::ResolvedStyle> SearchFieldResultsButtonElement::resolveCust
     if (!shadowHostStyle)
         return std::nullopt;
 
-    auto appearance = shadowHostStyle->effectiveAppearance();
-    if (appearance == StyleAppearance::TextField) {
+    auto effectiveAppearance = shadowHostStyle->effectiveAppearance();
+    if (effectiveAppearance == StyleAppearance::TextField && shadowHostStyle->appearance() == StyleAppearance::TextField) {
         auto elementStyle = resolveStyle(resolutionContext);
         elementStyle.style->setDisplay(DisplayType::None);
         return elementStyle;
     }
-    if (appearance != StyleAppearance::SearchField) {
+    if (effectiveAppearance != StyleAppearance::SearchField && effectiveAppearance != StyleAppearance::TextField) {
```

Comment:
These checks are not super pretty nor self explanatory IMO.

We should probably have a static method in this file that takes a RenderStyle and says whether the effectiveAppearance is default based on the writing mode and the effectiveAppearance.

Reply:
I'll add a static method with a good name for clarity. Though the method won't consider `writing-mode`, since that's a macOS specific behavior. It's more about the original appearance the author specified.